src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ScreenRecorder.cs

```
@@ -100,18 +103,92 @@ private void OnDestroy()
+                // If the gameplay session was not created, return
+                if (gameplaySessionId == -1)
                 {
-                    RGDebug.LogInfo($"Zipping recording replay to file: {_currentVideoDirectory}.zip");
-                    ZipFile.CreateFromDirectory(theVideoDirectory, theVideoDirectory + ".zip");
-                    Directory.Delete(theVideoDirectory, true);
-                    RGDebug.LogInfo($"Finished zipping replay to file: {_currentVideoDirectory}");
-                });
-                _isRecording = false;
+                    return;
+                }
+                
+                // Upload the gameplay session data
+                await _rgServiceManager.UploadGameplaySessionData(gameplaySessionId,
+                    _currentGameplaySessionDirectoryPrefix + "_data.zip",
+                    () => { RGDebug.LogInfo($"Uploaded gameplay session data"); },
+                    () => { RGDebug.LogError($"Failed to upload gameplay session data"); });
+                
+                // Next, upload the gameplay session screenshots
+                await _rgServiceManager.UploadGameplaySessionScreenshots(gameplaySessionId,
+                    _currentGameplaySessionDirectoryPrefix + "_screenshots.zip",
+                    () => { RGDebug.LogInfo($"Uploaded gameplay session screenshots"); },
+                    () => { RGDebug.LogError($"Failed to upload gameplay session screenshots"); });
+                
+                // Finally, upload the thumbnail
+                await _rgServiceManager.UploadGameplaySessionThumbnail(gameplaySessionId,
+                    _currentGameplaySessionDirectoryPrefix + "_thumbnail.png",
+                    () => { RGDebug.LogInfo($"Uploaded gameplay session thumbnail"); },
+                    () => { RGDebug.LogError($"Failed to upload gameplay session thumbnail"); });
+            }
+            catch (Exception e)
+            {
+                Debug.LogError(e);
```

Why are we hiding this exception ?  Use RGDebug and LogException, or take some code action here